### ISSUE TITLE
Segmentation fault MacOS 11.7.10 Moonlight 5

### ISSUE THREAD

**Issue by neatnoise** [moonlight-stream/moonlight-qt#1096]
**Describe the bug**
Segmentation fault on MacOS 11.7.10. Hardware Macbook Air Early 2014
```
tm-lap:Preferences tomek$ /Applications/Moonlight.app/Contents/MacOS/Moonlight 
Redirecting log output to /tmp/Moonlight-1698014573.log
Segmentation fault: 11
```
Removing config (plist) results the same
**Steps to reproduce**
Segmentation fault while launching the app Moonlight 5.0.0

**Screenshots**
![Screenshot 2023-10-23 at 00 38 21](https://github.com/moonlight-stream/moonlight-qt/assets/2226294/060beae6-e422-4539-b83b-88be289d9721)

**Affected games**
Core Moonlight functionality

**Other Moonlight clients**
- Does the issue occur when using Moonlight on iOS or Android?
No, only in the new version
**Moonlight settings (please complete the following information)**
Occurs with the current config and default (Removing config (plist) results the same).

**Gamepad-related issues (please complete if problem is gamepad-related)**
-

**Client PC details (please complete the following information)**
 - OS: MacOS 11.7.10
 - Moonlight Version: 5.0.0 
 - GPU: Intel GPU Macbook Air Early 2014

**Server PC details (please complete the following information)**
 - OS: Windows 10, up to date
 - Latest sunshine
 - GPU: AMD RX 580

**Moonlight Logs (please attach)**
Before segmentation fault no log is saved (screenshot)

**Additional context**
-
Adding ldd info:
```
otool -L /Applications/Moonlight.app/Contents/MacOS/Moonlight 
/Applications/Moonlight.app/Contents/MacOS/Moonlight:
	@loader_path/../Frameworks/libavcodec.59.dylib (compatibility version 59.0.0, current version 59.37.100)
	@loader_path/../Frameworks/libavutil.57.dylib (compatibility version 57.0.0, current version 57.28.100)
	@rpath/SDL2.framework/Versions/A/SDL2 (compatibility version 1.0.0, current version 2401.0.0)
	@rpath/SDL2_ttf.framework/Versions/A/SDL2_ttf (compatibility version 15.0.0, current version 15.1.0)
	/usr/lib/libobjc.A.dylib (compatibility version 1.0.0, current version 228.0.0)
	/System/Library/Frameworks/VideoToolbox.framework/Versions/A/VideoToolbox (compatibility version 1.0.0, current version 1.0.0)
	/System/Library/Frameworks/AVFoundation.framework/Versions/A/AVFoundation (compatibility version 1.0.0, current version 2.0.0)
	/System/Library/Frameworks/CoreVideo.framework/Versions/A/CoreVideo (compatibility version 1.2.0, current version 1.5.0)
	/System/Library/Frameworks/CoreGraphics.framework/Versions/A/CoreGraphics (compatibility version 64.0.0, current version 1557.5.4)
	/System/Library/Frameworks/CoreMedia.framework/Versions/A/CoreMedia (compatibility version 1.0.0, current version 1.0.0)
	/System/Library/Frameworks/CoreAudio.framework/Versions/A/CoreAudio (compatibility version 1.0.0, current version 1.0.0)
	/System/Library/Frameworks/AudioUnit.framework/Versions/A/AudioUnit (compatibility version 1.0.0, current version 1.0.0)
	@rpath/QtQuickControls2.framework/Versions/A/QtQuickControls2 (compatibility version 6.0.0, current version 6.4.0)
	@rpath/QtQuick.framework/Versions/A/QtQuick (compatibility version 6.0.0, current version 6.4.0)
	@rpath/QtOpenGL.framework/Versions/A/QtOpenGL (compatibility version 6.0.0, current version 6.4.0)
	@rpath/QtSvg.framework/Versions/A/QtSvg (compatibility version 6.0.0, current version 6.4.0)
	@rpath/QtGui.framework/Versions/A/QtGui (compatibility version 6.0.0, current version 6.4.0)
	/System/Library/Frameworks/AppKit.framework/Versions/C/AppKit (compatibility version 45.0.0, current version 2113.40.126)
	/System/Library/Frameworks/ImageIO.framework/Versions/A/ImageIO (compatibility version 1.0.0, current version 1.0.0)
	/System/Library/Frameworks/Metal.framework/Versions/A/Metal (compatibility version 1.0.0, current version 261.13.0)
	@rpath/QtQmlModels.framework/Versions/A/QtQmlModels (compatibility version 6.0.0, current version 6.4.0)
	@rpath/QtQml.framework/Versions/A/QtQml (compatibility version 6.0.0, current version 6.4.0)
	@rpath/QtNetwork.framework/Versions/A/QtNetwork (compatibility version 6.0.0, current version 6.4.0)
	@rpath/QtCore.framework/Versions/A/QtCore (compatibility version 6.0.0, current version 6.4.0)
	/System/Library/Frameworks/IOKit.framework/Versions/A/IOKit (compatibility version 1.0.0, current version 275.0.0)
	/System/Library/Frameworks/DiskArbitration.framework/Versions/A/DiskArbitration (compatibility version 1.0.0, current version 1.0.0)
	/System/Library/Frameworks/AGL.framework/Versions/A/AGL (compatibility version 1.0.0, current version 1.0.0)
	/System/Library/Frameworks/OpenGL.framework/Versions/A/OpenGL (compatibility version 1.0.0, current version 1.0.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1300.23.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.100.3)
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1858.112.0)
	/System/Library/Frameworks/CoreServices.framework/Versions/A/CoreServices (compatibility version 1.0.0, current version 1141.1.0)
	/System/Library/Frameworks/Foundation.framework/Versions/C/Foundation (compatibility version 300.0.0, current version 1858.112.0)
```

**Comment by neatnoise** [comment]
Adding ldd info:
```
otool -L /Applications/Moonlight.app/Contents/MacOS/Moonlight 
/Applications/Moonlight.app/Contents/MacOS/Moonlight:
	@loader_path/../Frameworks/libavcodec.59.dylib (compatibility version 59.0.0, current version 59.37.100)
	@loader_path/../Frameworks/libavutil.57.dylib (compatibility version 57.0.0, current version 57.28.100)
	@rpath/SDL2.framework/Versions/A/SDL2 (compatibility version 1.0.0, current version 2401.0.0)
	@rpath/SDL2_ttf.framework/Versions/A/SDL2_ttf (compatibility version 15.0.0, current version 15.1.0)
	/usr/lib/libobjc.A.dylib (compatibility version 1.0.0, current version 228.0.0)
	/System/Library/Frameworks/VideoToolbox.framework/Versions/A/VideoToolbox (compatibility version 1.0.0, current version 1.0.0)
	/System/Library/Frameworks/AVFoundation.framework/Versions/A/AVFoundation (compatibility version 1.0.0, current version 2.0.0)
	/System/Library/Frameworks/CoreVideo.framework/Versions/A/CoreVideo (compatibility version 1.2.0, current version 1.5.0)
	/System/Library/Frameworks/CoreGraphics.framework/Versions/A/CoreGraphics (compatibility version 64.0.0, current version 1557.5.4)
	/System/Library/Frameworks/CoreMedia.framework/Versions/A/CoreMedia (compatibility version 1.0.0, current version 1.0.0)
	/System/Library/Frameworks/CoreAudio.framework/Versions/A/CoreAudio (compatibility version 1.0.0, current version 1.0.0)
	/System/Library/Frameworks/AudioUnit.framework/Versions/A/AudioUnit (compatibility version 1.0.0, current version 1.0.0)
	@rpath/QtQuickControls2.framework/Versions/A/QtQuickControls2 (compatibility version 6.0.0, current version 6.4.0)
	@rpath/QtQuick.framework/Versions/A/QtQuick (compatibility version 6.0.0, current version 6.4.0)
	@rpath/QtOpenGL.framework/Versions/A/QtOpenGL (compatibility version 6.0.0, current version 6.4.0)
	@rpath/QtSvg.framework/Versions/A/QtSvg (compatibility version 6.0.0, current version 6.4.0)
	@rpath/QtGui.framework/Versions/A/QtGui (compatibility version 6.0.0, current version 6.4.0)
	/System/Library/Frameworks/AppKit.framework/Versions/C/AppKit (compatibility version 45.0.0, current version 2113.40.126)
	/System/Library/Frameworks/ImageIO.framework/Versions/A/ImageIO (compatibility version 1.0.0, current version 1.0.0)
	/System/Library/Frameworks/Metal.framework/Versions/A/Metal (compatibility version 1.0.0, current version 261.13.0)
	@rpath/QtQmlModels.framework/Versions/A/QtQmlModels (compatibility version 6.0.0, current version 6.4.0)
	@rpath/QtQml.framework/Versions/A/QtQml (compatibility version 6.0.0, current version 6.4.0)
	@rpath/QtNetwork.framework/Versions/A/QtNetwork (compatibility version 6.0.0, current version 6.4.0)
	@rpath/QtCore.framework/Versions/A/QtCore (compatibility version 6.0.0, current version 6.4.0)
	/System/Library/Frameworks/IOKit.framework/Versions/A/IOKit (compatibility version 1.0.0, current version 275.0.0)
	/System/Library/Frameworks/DiskArbitration.framework/Versions/A/DiskArbitration (compatibility version 1.0.0, current version 1.0.0)
	/System/Library/Frameworks/AGL.framework/Versions/A/AGL (compatibility version 1.0.0, current version 1.0.0)
	/System/Library/Frameworks/OpenGL.framework/Versions/A/OpenGL (compatibility version 1.0.0, current version 1.0.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1300.23.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.100.3)
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1858.112.0)
	/System/Library/Frameworks/CoreServices.framework/Versions/A/CoreServices (compatibility version 1.0.0, current version 1141.1.0)
	/System/Library/Frameworks/Foundation.framework/Versions/C/Foundation (compatibility version 300.0.0, current version 1858.112.0)
```

**Comment by neatnoise** [comment]
> > Adding ldd info:
> > ```
> > removed to save space here
> > ```
> 
> Just a heads-up: You may want to put this in the original post, and then delete the comment to keep it nice and clean. You don't have to, however I recommend you do so it's all in one post.
> 
> Also, what's the output when running Moonlight's executable with `sudo`?

With sudo the result is the same

```
Redirecting log output to /tmp/Moonlight-1698102347.log
Segmentation fault: 11
```

**Comment by Smitts148** [comment]
Getting the same error on 10.15.7  (Catalina)

**Comment by rjbritt** [comment]
Just for additional data, I'm seeing the same thing. 10.15.7, 2013 iMac 27"

**Comment by neatnoise** [comment]
> Did you download the Universal macOS version?

yes, universal package. There isn't other download version fo MacOS

**Comment by cgutman** [comment]
This is addressed in v5.0.1

**Comment by S0ulM1ke** [comment]
But what have changed? What was the issue?
